sdk/identity/azure-identity/src/client_certificate_credential.cpp

```diff
@@ -50,6 +60,204 @@ template <typename T> std::vector<uint8_t> ToUInt8Vector(T const& in)
   return outVec;
 }
 
+using CertificateThumbprint = std::vector<unsigned char>;
+using UniquePrivateKey = Azure::Identity::_detail::UniquePrivateKey;
+using PrivateKey = decltype(std::declval<UniquePrivateKey>().get());
+
+#ifdef AZ_PLATFORM_WINDOWS
+enum PrivateKeyType
+{
+  Rsa,
+  Ecdsa,
+  Pkcs
+};
+
+std::vector<uint8_t> PemToBinary(LPCSTR str, DWORD count)
+{
+  DWORD size = 0;
+  THROW_IF_WIN32_BOOL_FALSE(CryptStringToBinaryA(
+      str, count, CRYPT_STRING_BASE64HEADER, nullptr, &size, nullptr, nullptr));
+  std::vector<uint8_t> buffer(size);
+  THROW_IF_WIN32_BOOL_FALSE(CryptStringToBinaryA(
+      str, count, CRYPT_STRING_BASE64HEADER, buffer.data(), &size, nullptr, nullptr));
+  return buffer;
+}
+
+CertificateThumbprint GetThumbprint(PCCERT_CONTEXT cert)
```

Comment:
Should this take a raw `PCCERT_CONTEXT` or a `wil::unique_cert_context const&`?

Reply:
That is completely fair.

In general we pass the unique pointers around our code because we've been trained to treat raw pointers as something suspicious (there's nothing that stops someone from calling delete on the raw pointer, destroying the unique_ptr). 

By passing as a `const unique_ptr&` there are no copies done and the caller cannot do something bad to the underlying pointer.